src/CFamily/CMake/ActiveConfigProvider.cs

```diff
@@ -0,0 +1,113 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using Newtonsoft.Json.Linq;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Integration.Helpers;
+
+namespace SonarLint.VisualStudio.CFamily.CMake
+{
+    internal interface IActiveConfigProvider
+    {
+        /// <summary>
+        /// Returns the name of the current configuration
+        /// </summary>
+        /// <param name="rootDirectory">The root directory for the solution</param>
+        string GetActiveConfig(string rootDirectory);
+    }
+
+    internal class ActiveConfigProvider : IActiveConfigProvider
+    {
+        private const string VSDefaultConfiguration = "x64-Debug";
+        private const string RelativeSettingsPath = ".vs\\ProjectSettings.json";
+        private const string ConfigPropertyName = "CurrentProjectSetting";
+
+        private readonly ILogger logger;
+        private readonly IFileSystem fileSystem;
+
+        public ActiveConfigProvider(ILogger logger) : this(logger, new FileSystem())
+        {
+        }
+
+        internal /* for testing */ ActiveConfigProvider(ILogger logger, IFileSystem fileSystem)
+        {
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this.fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
+        }
+
+        /// <summary>
+        /// Reads the config from the internal VS CMake settings file
+        /// </summary>
+        /// <returns>The active config, or the default config if the setting could not be found/read</returns>
+        public string GetActiveConfig(string rootDirectory)
+        {
+            if (rootDirectory == null)
+            {
+                throw new ArgumentNullException(nameof(rootDirectory));
+            }
+
+            var fullPath = Path.Combine(rootDirectory, RelativeSettingsPath);
+            LogDebug("Internal VS CMake settings file: " + fullPath);
```

Comment:
I made all of the log message debug since they will be written for every analysis.
I see the compilation db locator ones are not debug. They should probably be consistent. What level do you think they should be?

Reply:
Oof, that's a good question. I haven't thought about the fact that it will run for every single analysis.
I'll change them to debug too.